pkg/upgrade/base.go

```diff
@@ -58,18 +56,6 @@ func newBase(log logr.Logger, config Config) (*base, error) {
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
-	clusterEndPoint := config.TargetCluster.TargetApiEndpoint
```

Comment:
This logic goes away. The user should specify the entire API endpoint & port. putting this burden on the user gives this function fewer things to do while making it more extensible. It now supports any port you like, not just a hardcoded 6443.